### PR TITLE
Doc for database managers installation

### PR DESCRIPTION
**Pull request by NovikovViktor**

### CODE REVIEW

**Comment by theofidry** [comment]
@NovikovViktor I think what you can specify in each part is the link towards the bundle installation documentation, f.e. http://symfony.com/doc/master/bundles/DoctrineBundle/index.html for DoctrineBundle, and the `doctrine/data-fixtures` package for all Doctrine flavours.

**Comment by NovikovViktor** [comment]
@theofidry, could you please check PR and give feedback

File: README.md [hunk 1]
```diff
@@ -38,6 +38,9 @@ Other references:
 
 ## Installation
 
+First you need install appropriate database managers (if you didn't installed it yet), according to your project requirements.
+How install supported database managers you can find [here](src/Resources/doc/database-managers-install.md).
```
Comment (theofidry) [inline README.md]: The second sentence is a bit redondant, might change this for: `Check the documentation [here](src/Resources/doc/database-managers-install.md).`

File: src/Resources/doc/database-managers-install.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Database managers
+
+## Doctrine ORM
+If you using relational databases (like MySql), you need to install:
```
Comment (theofidry) [inline src/Resources/doc/database-managers-install.md]: no need to bother to specify this. Le the user decide if he needs to use Doctrine ORM or not :)

File: src/Resources/doc/database-managers-install.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+1. [DoctrineORM](http://doctrine-orm.readthedocs.org/projects/doctrine-orm/en/latest/reference/configuration.html);
```
Comment (theofidry) [inline src/Resources/doc/database-managers-install.md]: Prefix the link by "install":

1. Install [DoctrineORM](http://doctrine-orm.readthedocs.org/projects/doctrine-orm/en/latest/reference/configuration.html)
2. Install [DoctrineBundle](http://symfony.com/doc/master/bundles/DoctrineBundle/installation.html)
3. Install [Doctrine Data Fixtures](https://github.com/doctrine/data-fixtures): `composer require --save-dev doctrine/data-fixtures`

**Comment by theofidry** [comment]
And apply my comments form `DoctrineORM` to the others and we're good :)

**Comment by NovikovViktor** [comment]
@theofidry, I added changes according to your comments, could you review it please

File: src/Resources/doc/database-managers-install.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Database managers
+
+## Doctrine ORM
+1. Install [DoctrineORM](http://doctrine-orm.readthedocs.org/projects/doctrine-orm/en/latest/reference/configuration.html);
+2. Install [DoctrineBundle](http://symfony.com/doc/master/bundles/DoctrineBundle/installation.html);
+3. Install [Doctrine Data Fixtures](https://github.com/doctrine/data-fixtures): composer require --save-dev doctrine/data-fixtures
```
Comment (theofidry) [inline src/Resources/doc/database-managers-install.md]: use the "`" character for the composer command.

File: src/Resources/doc/database-managers-install.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+## Doctrine ODM
+1. Install [DoctrineODM](http://doctrine-orm.readthedocs.org/projects/doctrine-mongodb-odm/en/latest/reference/introduction.html#setup)
+2. Install [DoctrineMongoDBBundle](http://symfony.com/doc/current/bundles/DoctrineMongoDBBundle/index.html#installation)
```
Comment (theofidry) [inline src/Resources/doc/database-managers-install.md]: Needs as 3. the step mentioned L6

File: src/Resources/doc/database-managers-install.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+## Doctrine PHPCR ODM
+1. Install [Jackalope Doctrine DBAL](https://github.com/jackalope/jackalope-doctrine-dbal#installation)
+2. Install [Doctrine PHPCR ODM](http://doctrine-orm.readthedocs.org/projects/doctrine-phpcr-odm/en/latest/reference/installation-configuration.html)
+3. Install [Bundle for integrate Doctrine PHPCR ODM with Symfony](http://symfony.com/doc/current/cmf/bundles/phpcr_odm/introduction.html)
```
Comment (theofidry) [inline src/Resources/doc/database-managers-install.md]: can replace the text `Bundle for integrate Doctrine PHPCR ODM with Symfony` by `DoctrinePHPCRBundle`

Comment (theofidry) [inline src/Resources/doc/database-managers-install.md]: Needs as 4. the step mentioned L6

File: src/Resources/doc/database-managers-install.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+After install [DoctrineFixturesBundle](http://symfony.com/doc/current/bundles/DoctrineFixturesBundle/index.html);
```
Comment (theofidry) [inline src/Resources/doc/database-managers-install.md]: No need for this bundle :) But you can change to:  `Then install [HautelookAliceBundle](https://github.com/hautelook/AliceBundle#installation)`

**Comment by theofidry** [comment]
Almost there :)

File: src/Resources/doc/database-managers-install.md (hunk 5 above)
Comment (NovikovViktor) [inline src/Resources/doc/database-managers-install.md]: @theofidry , you mean Install [DoctrineMongoDBBundle] should be third in list? and befor this insert:
3. Install [Doctrine Data Fixtures](https://github.com/doctrine/data-fixtures): composer require --save-dev doctrine/data-fixtures
right?

**Comment by NovikovViktor** [comment]
@theofidry , I added new changes except part with lines, need your clarification. Thanks :)

File: src/Resources/doc/database-managers-install.md (hunk 5 above)
Comment (theofidry) [inline src/Resources/doc/database-managers-install.md]: I mean the step `nstall [Doctrine Data Fixtures](https://github.com/doctrine/data-fixtures): composer require --save-dev doctrine/data-fixtures` should be added for each: Doctrie ORM, Doctrine ODM and Doctrine PHPCR.

File: src/Resources/doc/database-managers-install.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# Database managers
+
+## Doctrine ORM
+1. Install [DoctrineORM](http://doctrine-orm.readthedocs.org/projects/doctrine-orm/en/latest/reference/configuration.html);
+2. Install [DoctrineBundle](http://symfony.com/doc/master/bundles/DoctrineBundle/installation.html);
+3. Install [Doctrine Data Fixtures](https://github.com/doctrine/data-fixtures): `composer require --save-dev doctrine/data-fixtures`
+
+## Doctrine ODM
+1. Install [DoctrineODM](http://doctrine-orm.readthedocs.org/projects/doctrine-mongodb-odm/en/latest/reference/introduction.html#setup)
+2. Install [DoctrineMongoDBBundle](http://symfony.com/doc/current/bundles/DoctrineMongoDBBundle/index.html#installation)
```
Comment (theofidry) [inline src/Resources/doc/database-managers-install.md]: You're lacking the step:

3. Install [Doctrine Data Fixtures](https://github.com/doctrine/data-fixtures): `composer require --save-dev doctrine/data-fixtures`

here too

File: src/Resources/doc/database-managers-install.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+## Doctrine PHPCR ODM
+1. Install [Jackalope Doctrine DBAL](https://github.com/jackalope/jackalope-doctrine-dbal#installation)
+2. Install [Doctrine PHPCR ODM](http://doctrine-orm.readthedocs.org/projects/doctrine-phpcr-odm/en/latest/reference/installation-configuration.html)
+3. Install [DoctrinePHPCRBundle](http://symfony.com/doc/current/cmf/bundles/phpcr_odm/introduction.html)
```
Comment (theofidry) [inline src/Resources/doc/database-managers-install.md]: You're lacking the step:

4. Install [Doctrine Data Fixtures](https://github.com/doctrine/data-fixtures): `composer require --save-dev doctrine/data-fixtures`

here too

**Comment by theofidry** [comment]
Sorry my previous comment was not clear enough

**Comment by NovikovViktor** [comment]
@theofidry , added that line to each block,  should I change order of items?

**Comment by theofidry** [comment]
The order is fine :)

File: src/Resources/doc/database-managers-install.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+2. Install [DoctrineMongoDBBundle](http://symfony.com/doc/current/bundles/DoctrineMongoDBBundle/index.html#installation)
+3. Install [Doctrine Data Fixtures](https://github.com/doctrine/data-fixtures): `composer require --save-dev doctrine/data-fixtures`
+
+## Doctrine PHPCR ODM
+1. Install [Jackalope Doctrine DBAL](https://github.com/jackalope/jackalope-doctrine-dbal#installation)
+2. Install [Doctrine PHPCR ODM](http://doctrine-orm.readthedocs.org/projects/doctrine-phpcr-odm/en/latest/reference/installation-configuration.html)
+3. Install [DoctrinePHPCRBundle](http://symfony.com/doc/current/cmf/bundles/phpcr_odm/introduction.html)
+4. Install [Doctrine Data Fixtures](https://github.com/doctrine/data-fixtures): `composer require --save-dev doctrine/data-fixtures`
+
+Then install [HautelookAliceBundle](https://github.com/hautelook/AliceBundle#installation)
```
Comment (theofidry) [inline src/Resources/doc/database-managers-install.md]: just a dot at the end

**Comment by theofidry** [comment]
Ok just the dot at the and and then rebase it to squash the commits. If you're no sure of how to do it: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by NovikovViktor** [comment]
@theofidry , please check if I did this right

**Comment by theofidry** [comment]
Note, it added commits instead :(

*. Do: `git rebase -i a810c303c2fbec1078020f7786ea27387b4bf6ab`
*. You should have a screen with 10 lines somewhat similar to:

```
pick <somecode> <some comment>
...
pick <somecode> <some comment>
```

* Starting the second line, replace all the `pick` by `fixup`
* If you do a `git log`, you should then see my latests commit and then yours (only 1 commit versus currently 9).
* If you wish to change your commit comment, then do: `git commit --amend`, and then change your commit message
* Repush

**Comment by NovikovViktor** [comment]
@theofidry, when I do this command I get message about conflicts, after  I run git rebase --continue, both files marked as should be merged. And in git log I already see commit with all my ten commits. So I need to merge and push or seomething going wrong?

**Comment by theofidry** [comment]
I would say that something when wrong :sweat_smile: 

Do the following:

* `git fetch --all`
* `git checkout origin/master`
* `git branch -D master`
* `git checkout -b master`

This will reset your branch to a "fresh" state. Then redo:

>* Do: `git rebase -i a810c303c2fbec1078020f7786ea27387b4bf6ab`
* You should have a screen with 10 lines somewhat similar to:
>
```
pick <somecode> <some comment>
...
pick <somecode> <some comment>
```
>
* Starting the second line, replace all the `pick` by `fixup`
* If you do a `git log`, you should then see my latests commit and then yours (only 1 commit versus currently 9).
* If you wish to change your commit comment, then do: `git commit --amend`, and then change your commit message
* Repush

**Comment by theofidry** [comment]
@NovikovViktor if you're still having trouble move to Gitter (chat room link on README)

**Comment by NovikovViktor** [comment]
@theofidry,  do like you say still have same problem. I did not see link to gitter in readme

**Comment by theofidry** [comment]
Sorry about that fixed

**Comment by NovikovViktor** [comment]
@theofidry , its ok, I am sorry for that confusing situation with git, I wrote to you in Gitter about the same error

**Comment by theofidry** [comment]
Closed in favour of #160.

**Pull request closed without merging**